tests/test_financial.py

```diff
@@ -90,13 +93,167 @@ def test_decimal_with_when(self):
 
 
 class TestPV:
+    # Test cases for pytest parametrized example-based tests
+    test_cases = {
+        "default_fv_and_when": {
+            "inputs": {
+                "rate": 0.05,
+                "nper": 10,
+                "pmt": 1000,
+            },
+            "expected_result": -7721.73,
+        },
+        "specify_fv_and_when": {
+            "inputs": {
+                "rate": 0.05,
+                "nper": 10,
+                "pmt": 1000,
+                "fv": 0,
+                "when": 0,
+            },
+            "expected_result": -7721.73,
+        },
+        "when_1": {
+            "inputs": {
+                "rate": 0.05,
+                "nper": 10,
+                "pmt": 1000,
+                "fv": 0,
+                "when": 1,
+            },
+            "expected_result": -8107.82,
+        },
+        "when_1_and_fv_1000": {
+            "inputs": {
+                "rate": 0.05,
+                "nper": 10,
+                "pmt": 1000,
+                "fv": 1000,
+                "when": 1,
+            },
+            "expected_result": -8721.73,
+        },
+        "fv>0": {
+            "inputs": {
+                "rate": 0.05,
+                "nper": 10,
+                "pmt": 1000,
+                "fv": 1000,
+            },
+            "expected_result": -8335.65,
+        },
+        "negative_rate": {
+            "inputs": {
+                "rate": -0.05,
+                "nper": 10,
+                "pmt": 1000,
+                "fv": 0,
+            },
+            "expected_result": -13403.65,
+        },
+        "rates_as_array": {
+            "inputs": {
+                "rate": numpy.array([0.010, 0.015, 0.020, 0.025, 0.030, 0.035]),
+                "nper": 10,
+                "pmt": 1000,
+                "fv": 0,
+            },
+            "expected_result": numpy.array(
+                [-9471.30, -9222.18, -8982.59, -8752.06, -8530.20, -8316.61]
+            ),
+        },
+    }
+
+    # Randomized input strategies for fuzz tests & property-based tests
```

Comment:
Providing a "strategy" limits the values that the fuzzer (Hypothesis) will supply. The strategies that we re-use are defined here.